File: repos_estudiantes_TARQUI_tercera_entrega/LUNA_CRUZ_CAMILA_MILENKA/PRACTICA 5-6/GENERICIDAD/EJERCICIO 1/Ejercicio1.java
//1. Crea una clase genérica Caja<T> para guardar algún tipo de objeto
public class Ejercicio1 {
    public static class Caja<T> {
        private T objeto;
// Método para guardar el objeto
    public void guardar(T objeto) {
        this.objeto = objeto;
    }
// Método para obtener el objeto
    public T obtener() {
        return this.objeto;
    }
}
public static void main(String[] args) {
//b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo
//c) Muestra el contenido de las cajas   
    Caja<String> cajaString = new Caja<>();
    cajaString.guardar("Fardo Americano");
    System.out.println("Contenido de caja 1: " + cajaString.obtener());

    Caja<Integer> cajaInteger = new Caja<>();
    cajaInteger.guardar(1250);
    System.out.println("Contenido de caja 2: " + cajaInteger.obtener() + "Bs");
  }
}

File: repos_estudiantes_TARQUI_tercera_entrega/LUNA_CRUZ_CAMILA_MILENKA/PRACTICA 5-6/GENERICIDAD/EJERCICIO 1/Ejercicio1.py
#1. Crea una clase genérica Caja<T> para guardar algún tipo de objeto
from typing import Generic, TypeVar
T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self):
        self.objeto: T = None 
#a) Agrega métodos guardar() y obtener()
    def guardar(self, objeto: T):
        self.objeto = objeto
    def obtener(self) -> T:
        return self.objeto

if __name__ == "__main__":
#b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo
    caja_string = Caja[str]()
    caja_string.guardar("Fardo Americano")
    print("Contenido de caja 1:", caja_string.obtener())
#c) Muestra el contenido de las cajas
    caja_entero = Caja[int]()
    caja_entero.guardar(1250)
    print("Contenido de caja 2:", caja_entero.obtener())